File: backend/internal/models/anime.go
package models

import "time"

type Anime struct {
	MALID int64 `json:"mal_id"`
	Title string `json:"title"`
	TitleEnglish *string `json:"title_english,omitempty"`
	Synopsis *string `json:"synopsis,omitempty"`
	Score *float64 `json:"score,omitempty"`
	Popularity *int `json:"popularity,omitempty"`
	Episodes *int  `json:"episodes,omitempty"`
	Year *int `json:"year,omitempty"`
	ImageURL *string `json:"image_url,omitempty"`
	GenresJSON []byte `json:"-"`
	StudiosJSON []byte `json:"-"`
	CreatedAt *time.Time `json:"created_at,omitempty"`
	UpdatedAt *time.Time `json:"created_at,omitempty"`
}

File: frontend/src/types/anime.ts
export type Anime = {
  mal_id: number;
  title: string;
  title_english?: string | null;
  synopsis?: string | null;
  score?: number | null;
  popularity?: number | null;
  episodes?: number | null;
  year?: number | null;
  image_url?: string | null;
};

export type Recommendation = {
  source_anime_id: number;
  recommended_anime_id: number;
  score: number;
  rank: number;
  reason: string;
  model_version: string;
  anime: Anime;
};